corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -82,10 +84,20 @@ hqDefine("data_dictionary/js/data_dictionary", [
                 );
                     groupObj.properties.push(propObj);
                 }
+                groupObj.properties.subscribe(self.changeSaveButton);
                 self.groups.push(groupObj);
             }
         };
 
+        self.fetchCaseProperties = function () {
+            if (self.groups().length === 0) {
```

Comment:
I admittedly don't have a lot of context around the data dictionary, hence me asking this: what is the group in this context and why do we check whether there is a group here?

Reply:
When the user selects a case type, the viewmodel fetches its case properties.

The server returns the CaseProperty instances nested in their CasePropertyGroup instances ("data.groups" on line 96). (The CaseProperty instances that don't have a CasePropertyGroup are nested in a group with an empty name.)

So if `self.groups()` does not have a value, it means that the viewmodel has not fetched its case properties already, so it should do that now.